test/prim.test.did

```diff
@@ -24,10 +24,125 @@ assert blob "DIDL\00\01\7e"                !: (bool) "bool: missing";
+assert blob "DIDL\00\01\76\ff\00" == "(255)"       : (int16) "int16: 255";
+assert blob "DIDL\00\01\76\00\01" == "(256)"       : (int16) "int16: 256";
+assert blob "DIDL\00\01\76\ff\ff" == "(-1)"        : (int16) "int16: -1";
+assert blob "DIDL\00\01\76"                       !: (int16) "int16: too short";
+assert blob "DIDL\00\01\76\00"                    !: (int16) "int16: too short";
+
+assert blob "DIDL\00\01\75\00\00\00\00" == "(0)"          : (int32) "int32: 0";
+assert blob "DIDL\00\01\75\01\00\00\00" == "(1)"          : (int32) "int32: 1";
+assert blob "DIDL\00\01\75\ff\00\00\00" == "(255)"        : (int32) "int32: 255";
+assert blob "DIDL\00\01\75\00\01\00\00" == "(256)"        : (int32) "int32: 256";
+assert blob "DIDL\00\01\75\ff\ff\00\00" == "(65535)"      : (int32) "int32: 65535";
+assert blob "DIDL\00\01\75\ff\ff\ff\ff" == "(-1)"         : (int32) "int32: -1";
+assert blob "DIDL\00\01\75"                              !: (int32) "int32: too short";
+assert blob "DIDL\00\01\75\00"                           !: (int32) "int32: too short";
+assert blob "DIDL\00\01\75\00\00"                        !: (int32) "int32: too short";
+assert blob "DIDL\00\01\75\00\00\00"                     !: (int32) "int32: too short";
+
+assert blob "DIDL\00\01\74\00\00\00\00\00\00\00\00" == "(0)"          : (int64) "int64: 0";
+assert blob "DIDL\00\01\74\01\00\00\00\00\00\00\00" == "(1)"          : (int64) "int64: 1";
+assert blob "DIDL\00\01\74\ff\00\00\00\00\00\00\00" == "(255)"        : (int64) "int64: 255";
+assert blob "DIDL\00\01\74\00\01\00\00\00\00\00\00" == "(256)"        : (int64) "int64: 256";
+assert blob "DIDL\00\01\74\ff\ff\00\00\00\00\00\00" == "(65535)"      : (int64) "int64: 65535";
+assert blob "DIDL\00\01\74\ff\ff\ff\ff\00\00\00\00" == "(4294967295)" : (int64) "int64: 4294967295";
+assert blob "DIDL\00\01\74\ff\ff\ff\ff\ff\ff\ff\ff" == "(-1)"         : (int64) "int64: -1";
+assert blob "DIDL\00\01\74"                              !: (int64) "int64: too short";
+assert blob "DIDL\00\01\74\00"                           !: (int64) "int64: too short";
+assert blob "DIDL\00\01\74\00\00"                        !: (int64) "int64: too short";
+assert blob "DIDL\00\01\74\00\00\00"                     !: (int64) "int64: too short";
+assert blob "DIDL\00\01\74\00\00\00\00"                  !: (int64) "int64: too short";
+assert blob "DIDL\00\01\74\00\00\00\00\00"               !: (int64) "int64: too short";
+assert blob "DIDL\00\01\74\00\00\00\00\00\00"            !: (int64) "int64: too short";
+assert blob "DIDL\00\01\74\00\00\00\00\00\00\00"         !: (int64) "int64: too short";
+
+// TODO: float
+
+
+assert blob "DIDL\00\01\71\00" == "(\"\")"             : (text) "text: empty string";
+assert blob "DIDL\00\01\71\06Motoko" == "(\"Motoko\")" : (text) "text: Motoko";
+assert blob "DIDL\00\01\71\05Motoko"                  !: (text) "text: too long";
+assert blob "DIDL\00\01\71\07Motoko"                  !: (text) "text: too short";
```

Comment:
Also add tests with overlong string length LEBs?

Reply:
Done, and also for the typ table and arg sequence length leb